### PR TITLE
[EAGLE-4201]: json logging

### PR DESCRIPTION
**Pull request by zeiler**
Added a json logger so it's convenient to get logs into logging stacks. Logs will look like: 

```
{"msg": "Loop iteration: 0 for thread 140485041844224", "@timestamp": "2024-09-24T21:31:15.411793Z", "filename": "test.py", "stack_info": null, "lineno": 214, "level": "info", "req_id": "runner-084a853704604c948bf226911f3cb0c6", "orig_req_id": "runner-084a853704604c948bf226911f3cb0c6"}
```
when env var `ENABLE_JSON_LOGGER=true`

### CODE REVIEW

File: clarifai/utils/logging.py [hunk 1]
```diff
@@ -151,3 +201,153 @@ def display_concept_relations_tree(relations_dict: Dict[str, Any]) -> None:
...
       tree.add(child)
     rprint(tree)
+
+
+def _default_json_default(obj):
+  """
+  Handle objects that could not be serialized to JSON automatically.
+
+  Coerce everything to strings.
+  All objects representing time get output as ISO8601.
```
Comment (wemoveon2) [inline clarifai/utils/logging.py:211]: this in UTC?

**Review by wemoveon2 — APPROVED** [review]
tests?

File: clarifai/utils/logging.py (hunk 1 above)
Comment (zeiler) [reply clarifai/utils/logging.py:211]: yeah i think so. `2024-09-24T21:31:15.411793Z`

File: tests/test_misc.py [hunk 2]
```diff
@@ -17,3 +19,49 @@ def test_get_logger_defaults():
   assert logger.level == logging.NOTSET
   assert logger.name == _get_library_name()
   assert isinstance(logger.handlers[0], RichHandler)
+
+
+def test_json_logger():
```
Comment (wemoveon2) [inline tests/test_misc.py:24]: did we not use pytest?

**Review by wemoveon2 — APPROVED** [review]

File: tests/test_misc.py (hunk 2 above)
Comment (zeiler) [reply tests/test_misc.py:24]: pytest picks these up.

**Pull request merged**